Use property classes in single table QualityReport

R-Palazzo opened the pull request:
Resolve #370.

amontanez24 commented on sdmetrics/reports/single_table/quality_report.py:
```diff
@@ -25,35 +18,71 @@ class QualityReport():
+        if missing_columns:
+            error_message = (
+                'The metadata does not match the data. The following columns are missing'
+                ' in the real/synnthetic data or in the metadata: '
```
typo: synnthetic

amontanez24 commented on sdmetrics/reports/single_table/quality_report.py:
```diff
@@ -68,42 +97,33 @@ def generate(self, real_data, synthetic_data, metadata, verbose=True):
         if verbose:
             self._print_results()
 
+    def _validate_property_generation(self, property_name):
```
minor: I'd rename to `_validate_property_generated`

amontanez24 commented on sdmetrics/reports/single_table/quality_report.py:
```diff
@@ -25,35 +18,71 @@ class QualityReport():
+            metadata (dict):
+                The metadata of the table.
+        """
+        missing_columns = set()
```
I don't think you need to define this here

amontanez24 commented on tests/integration/reports/single_table/test_single_table_quality_report.py:
```diff
@@ -1,59 +1,273 @@
+            report.get_details('Column Pair Trends'), expected_details_cpt
+        )
+
+    def test_report_end_to_end_with_errors(self):
```
minor: missing docstring

amontanez24 commented on tests/integration/reports/single_table/test_single_table_quality_report.py:
```diff
@@ -1,59 +1,273 @@
+            'Score': [0.750000, 0.5],
+        }))
+
+    def test_report_end_to_end(self):
```
missing docstring

amontanez24 commented on tests/unit/reports/single_table/test_single_table_quality_report.py:
```diff
@@ -1,455 +1,223 @@
+        )
 
-        Expect that the overall quality score is returned.
+    def test__validate_property_generation(self):
```
If you update the method name update it here too please

amontanez24 reviewed (commented):
Mostly minor comments. This is looking great! Things are really coming together

R-Palazzo commented:
Thanks for the review @amontanez24. I addressed the comments in [eb364fe](https://github.com/sdv-dev/SDMetrics/pull/377/commits/eb364fee2d91b136254dd79b3b35d6e180d3f272)

fealho reviewed (approved):
💯
